src/software/ai/passing/cost_function.cpp

```diff
@@ -31,14 +35,24 @@ double ratePass(const World& world, const Pass& pass, const Rectangle& zone,
 
     double in_region_quality = rectangleSigmoid(zone, pass.receiverPoint(), 0.2);
 
-    // Place strict limits on the ball speed
-    double min_pass_speed     = passing_config.min_pass_speed_m_per_s();
-    double max_pass_speed     = passing_config.max_pass_speed_m_per_s();
-    double pass_speed_quality = sigmoid(pass.speed(), min_pass_speed, 0.2) *
-                                (1 - sigmoid(pass.speed(), max_pass_speed, 0.2));
-
     return static_pass_quality * friendly_pass_rating * enemy_pass_rating *
-           shoot_pass_rating * pass_speed_quality * in_region_quality;
+           pass_backwards_rating * shoot_pass_rating * in_region_quality;
+}
+
+double ratePassBackwardsQuality(const Field& field, const Pass& pass,
+                                TbotsProto::PassingConfig& passing_config)
+{
+    if (field.pointInFriendlyHalf(pass.receiverPoint()) &&
+        field.pointInEnemyHalf(pass.passerPoint()))
+    {
+        double pass_distance = (pass.receiverPoint() - pass.passerPoint()).length();
+        if (pass_distance > passing_config.backwards_pass_distance())
+        {
+            return 0;
+        }
+    }
+
+    return 1;
```

Comment:
I think it would be better to approach 0 as the pass distance increases instead of straight up returning a 0. There may be some times when a long backwards pass is actually a good option (maybe its the only way to avoid turning over possession), so the other cost functions might have very high ratings that would be annihilated by this cost function returning 0.